src/zh/manxiaosi/src/eu/kanade/tachiyomi/extension/zh/manxiaosi/ManXiaoSi.kt

```diff
@@ -0,0 +1,325 @@
+                            genreList.add(
+                                it.text().trim(),
+                            )
+                        }
+                    }
+
+                    else -> {}
+                }
+            }
+            genre = genreList.joinToString()
+        }
+    }
+
+    // chapterList
+    override fun chapterListParse(response: Response): List<SChapter> {
+        val document = response.asJsoup()
+
+        val lis = mutableListOf<SChapter>()
+        document.select("#detail-list-select > li > a").forEach {
+            lis.add(
+                SChapter.create().apply {
+                    url = it.attr("href")
+                    name = it.text()
+                },
+            )
+        }
+        return lis
+    }
+
+    // image
+
+    override fun imageRequest(page: Page): Request {
+        return GET("$baseUrl${page.imageUrl}", headers)
+    }
+
+    override fun imageUrlParse(response: Response): String = throw UnsupportedOperationException()
+
+    // pageList
+
+    override fun pageListParse(response: Response): List<Page> {
+        val document = response.asJsoup()
+
+        val lis = mutableListOf<Page>()
+        document.select("#content > div.comiclist > div.comicpage > div > img")
+            .forEachIndexed { index, it ->
+                val url = it.attr("data-original")
+                lis.add(Page(index, imageUrl = url.toHttpUrl().encodedPath))
+            }
+        return lis
+    }
+
+    override fun popularMangaParse(response: Response): MangasPage {
+        val document = response.asJsoup()
+
+        val lis = mutableListOf<SManga>()
+        document.select("div.clearfix > div > div > h2 > a , div.mh-itme-top > a ").forEach {
+            lis.add(
+                SManga.create().apply {
+                    url = it.attr("href")
+                    title = it.attr("title")
+                    thumbnail_url = "$baseUrl/static/upload$url/cover.jpg"
+                },
+            )
+        }
+        return MangasPage(lis, false)
+    }
+
+    override fun popularMangaRequest(page: Int): Request {
+        return GET("$baseUrl/rank", headers)
+    }
+
+    override fun searchMangaParse(response: Response): MangasPage {
+        val document = response.asJsoup()
+
+        val lis = mutableListOf<SManga>()
+        document.select("ul > li > div.mh-item > a").forEach {
+            lis.add(
+                SManga.create().apply {
+                    url = it.attr("href")
+                    title = it.attr("title")
+                    thumbnail_url = "$baseUrl/static/upload$url/cover.jpg"
+                },
+            )
+        }
+
+        return MangasPage(lis, document.selectFirst("#nextPage") != null)
+    }
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        return if (query.isNotEmpty()) {
+            GET("$baseUrl/search?keyword=${query.trim()}", headers)
```

Comment:
Let the OkHttp library encode the query properly.

```suggestion
            val url = "$baseUrl/search".toHttpUrl().newBuilder()
                .addQueryParameter("keyword", query.trim())
                .build()
            GET(url, headers)
```